### ISSUE TITLE
Missing quantity values for SLE project

### ISSUE THREAD

**Issue by ColinHDev** [OpenEnergyPlatform/oeo-extended#13]
For the SLE project, OEO(X) concepts for the following quantity values are needed:

- population density [Inhabitants / km²]
- energy produced from renewable energies per inhabitant [MWh / Inhabitant]
- energy produced from renewable energies per area [MWh / km²]
- power capacity from renewable energies per area [MW / km²]
- wind turbine count per area [1 / km²]
- electricity demand per inhabitant [kWh / Inhabitant]
- heat demand per inhabitant [kWh / Inhabitant]
- battery count [1]
- battery energy storage content [GWh]

This would be a first concept for the taxonomies of the new concepts:
![image](https://github.com/user-attachments/assets/a9e39a5e-74d7-427d-b0f8-8ad08f762def)

- Most concepts describe something per area or inhabitant, so it made sense to group them under a parent class: `areal quantity value` / `populational quantity value`
  - There were already the `areal energy density` and the `areal power density` which described concepts per a certain area. But they are implemented in the OEO, so at least the `areal quantity value` parent class could not be implemented in OEOX.
- Same for the `unit`s, as it made sense to have a common parent class with a denominator of some `area unit`.
  - The `areal energy density unit` and `areal power density unit` are again implemented in the OEO, so the `areal unit` could not be implemented in OEOX. 
- I wasn't sure how to model "per inhabitant" as a unit. I would have liked to create something similar to `areal unit`, but I wasn't sure what the denominator would be.

**Comment by stap-m** [comment]
> * There were already the `areal energy density` and the `areal power density` which described concepts per a certain area. But they are implemented in the OEO, so at least the `areal quantity value` parent class could not be implemented in OEOX.

i'd prefer to avoid another layer. These classes could also be grouped by other criteria.

Here some remarks:
- I just saw that `areal energy density/power unit`, although in oeo, could use an equivalent axiom to describe the nominator/denominator.
- `areal count unit` could have the alternative label `count per area unit`
- `wind turbine count per area` could have the alternative label `number of wind turbines per area`
- `energy produced from renewables per area`: the unit here is also `areal energy density` :)
- `power capacity from renewables per area`: the unit here is also `areal power density` :)

Could you please create the equivalece axioms for these classes?

> * I wasn't sure how to model "per inhabitant" as a unit. I would have liked to create something similar to `areal unit`, but I wasn't sure what the denominator would be.

The unit of e.g. "energy per inhabitant" is just an energy unit (i.e. the denominator is 1).

**Comment by ColinHDev** [comment]
> * I just saw that `areal energy density/power unit`, although in oeo, could use an equivalent axiom to describe the nominator/denominator.

Yes, but I think that's the case for all the OEO-implemented subclasses of `unit`, like `square kilometer`, `area per power unit`, ...
But since the object properties `has unit numerator` and `has unit denominator`, they are technically also missing on all the composed units of UO, like `watt-hour`

> * `energy produced from renewables per area`: the unit here is also `areal energy density` :)
> * `power capacity from renewables per area`: the unit here is also `areal power density` :)

Ahh, apparently I misunderstood their definition. But while rereading I noticed the definition of `areal solar energy density` as a subclass of `areal energy density`: _An areal solar energy density is an areal energy density that gives the arriving solar power per area. A synonym for areal solar power density is irradiation._
Shouldn't the 2 "_power_" in there be swapped for "_energy_"?

Also, while being almost finished with this and working on the axioms, I noticed there is also an `irradiance unit` by the UO, which would make `areal power density unit` obsolete? 

> Could you please create the equivalece axioms for these classes?

First time creating these axioms, so there's probably still room for improvement:
```
'areal energy density unit'
    EquivalentTo 'unit'
        and 'has linear unit numerator' some 'energy unit'
        and 'has linear unit denominator' some 'area unit'

'areal power density unit' 
    EquivalentTo 'unit'
        and 'has linear unit numerator' some 'power unit'
        and 'has linear unit denominator' some 'area unit'

'areal count unit' 
    EquivalentTo 'unit'
        and 'has linear unit numerator' some count unit
        and 'has linear unit denominator' some area unit

'areal population count unit' 
    EquivalentTo 'areal count unit' 
        and 'has linear unit numerator' some 'population count unit'

'energy produced from renewable energies per area'
    EquivalentTo 'quantity value' 
        and 'is about' some 'renewable energy'
        and 'is about' some 'two-dimensional spatial region'

'power capacity from renewable energies per area'
    EquivalentTo 'quantity value' 
        and 'is about' some 'renewable energy'
        and 'is about' some 'two-dimensional spatial region'

'wind turbine count per area'
    EquivalentTo 'quantity value' 
        and 'quantity value of' some 'wind energy converting unit'
        and 'is about' some 'two-dimensional spatial region'

'population density' 
    EquivalentTo 'quantity value' 
        and 'quantity value of' some 'population'
        and 'is about' some 'two-dimensional spatial region'

'battery count'
    EquivalentTo 'quantity value' 
        and 'quantity value of' some 'battery'

'battery energy storage content'
    EquivalentTo 'energy storage content' 
        and 'quantity value of' some 'battery'

'energy produced from renewable energies per inhabitant'
    EquivalentTo 'quantity value' 
        and 'is about' some 'renewable energy'

'electricity demand per inhabitant'
    EquivalentTo 'quantity value' 
        and 'quantity value of' some 'electricity demand'

'heat demand per inhabitant'
    EquivalentTo 'quantity value' 
        and 'quantity value of' some 'energy demand'
        and 'is about' some 'thermal energy'
```

> The unit of e.g. "energy per inhabitant" is just an energy unit (i.e. the denominator is 1).

In the meeting, we discussed implementing a special `population count unit` as a subclass of `count unit`. I want to make sure, that thought is wrong: Saying our `electricity demand per inhabitant` has a unit with a denominator of `population count unit` doesn't work, because we don't want our class to represent the electricity demand of some number of people, like the electricity demand per city (with obviously varying population counts per city), but the electricity demand being for a single inhabitant making the denominator `1`?

This would be the current version:
![image](https://github.com/user-attachments/assets/12695e24-d0bb-42ee-a69f-e7fff599cb29)

**Comment by stap-m** [comment]
> > * `energy produced from renewables per area`: the unit here is also `areal energy density` :)
> > * `power capacity from renewables per area`: the unit here is also `areal power density` :)
> 
> Ahh, apparently I misunderstood their definition. But while rereading I noticed the definition of `areal solar energy density` as a subclass of `areal energy density`: _An areal solar energy density is an areal energy density that gives the arriving solar power per area. A synonym for areal solar power density is irradiation._ Shouldn't the 2 "_power_" in there be swapped for "_energy_"?

Well spotted, thanks. I am not 100% sure whether this is a bug. Could cou open an issue in oeo repo on this please?

> Also, while being almost finished with this and working on the axioms, I noticed there is also an `irradiance unit` by the UO, which would make `areal power density unit` obsolete?

Hm, we should check the def, if it is actually the same thing. If yes, we can add a mapping annotation. I don't want to make our class obsolete.
However, this is another hint that we have to do something about the units. Maybe the first stept is to get rid of the redundant stuff... Could you please also open a separate issue? This might require some good preparation and then a discussion in a dev-meeting.

**Comment by stap-m** [comment]
> In the meeting, we discussed implementing a special `population count unit` as a subclass of `count unit`. I want to make sure, that thought is wrong: Saying our `electricity demand per inhabitant` has a unit with a denominator of `population count unit` doesn't work, because we don't want our class to represent the electricity demand of some number of people, like the electricity demand per city (with obviously varying population counts per city), but the electricity demand being for a single inhabitant making the denominator `1`?

Exactly. 
A `population count unit` would make sense for something like `inhabitants per km2`. But I guess the `count unit` works here, too, right?

**Comment by ColinHDev** [comment]
> Well spotted, thanks. I am not 100% sure whether this is a bug. Could cou open an issue in oeo repo on this please?

Done in [#1981](https://github.com/OpenEnergyPlatform/ontology/issues/1981)

> Hm, we should check the def, if it is actually the same thing. If yes, we can add a mapping annotation. I don't want to make our class obsolete.
However, this is another hint that we have to do something about the units. Maybe the first stept is to get rid of the redundant stuff... Could you please also open a separate issue? This might require some good preparation and then a discussion in a dev-meeting.

Done in [#1982](https://github.com/OpenEnergyPlatform/ontology/issues/1982)

> Exactly. A `population count unit` would make sense for something like `inhabitants per km2`. But I guess the `count unit` works here, too, right?

So `electricity demand per inhabitant` having `count unit` as the denominator?

**Comment by ColinHDev** [comment]
As discussed with @stap-m, the units of `electricity demand per inhabitant`, `heat demand per inhabitant`, ... should represent that they don't just represent some energy value but always in regard to part of the population. So although the denominator technically is `1`, the following implementation has `population count unit` as the denominator. 

![image](https://github.com/user-attachments/assets/6fdc6426-3c31-49ea-873a-ae64e162314e)
